src/modules/python/pbs/v1/_pmi_utils.py

```diff
@@ -0,0 +1,158 @@
+"""
+
+/* 
+#  Copyright (C) 1994-2017 Altair Engineering, Inc.
+#  For more information, contact Altair at www.altair.com.
+#   
+#  This file is part of the PBS Professional ("PBS Pro") software.
+#  
+#  Open Source License Information:
+#   
+#  PBS Pro is free software. You can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License as published by the Free 
+#  Software Foundation, either version 3 of the License, or (at your option) any 
+#  later version.
+#   
+#  PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY 
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+#  PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+#   
+#  You should have received a copy of the GNU Affero General Public License along 
+#  with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   
+#  Commercial License Information: 
+#  
+#  The PBS Pro software is licensed under the terms of the GNU Affero General 
+#  Public License agreement ("AGPL"), except where a separate commercial license 
+#  agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+#   
+#  Altair’s dual-license business model allows companies, individuals, and 
+#  organizations to create proprietary derivative works of PBS Pro and distribute 
+#  them - whether embedded or bundled with other software - under a commercial 
+#  license agreement.
+#  
+#  Use of Altair’s trademarks, including but not limited to "PBS™", 
+#  "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's 
+#  trademark licensing policies.
+ *
+ */
+"""
+__doc__ = """
+$Id$
+Utility power functions.
+"""
+
+import pbs
+import os
+import sys
+
+
+def _pbs_conf(confvar):
+    # Return the value of a setting in the pbs.conf file if it exists.
+    # Save the values in a global dictionary for future use.
+
+    if confvar in os.environ:
+        return os.environ[confvar]
+
+    global pmi_pbsconf
+    if "pmi_pbsconf" not in globals():
+        pmi_pbsconf = dict()
+        cfile = "PBS_CONF_FILE"
+        if cfile in os.environ:
+            pbsconf = os.environ[cfile]
+        else:
+            pbsconf = "/etc/pbs.conf"
+
+        try:
+            fp = open(pbsconf)
+        except:
+            pbs.logmsg(pbs.DEBUG, "%s: conf file not found" % pbsconf)
```

Comment:
Unless we are sure that the file is not present, I'd say just log "unable to open <filename>"